src/lib/node_status_service/node_status_service.ml

```diff
@@ -0,0 +1,396 @@
+      0. ;
+    Gauge.set
+      (snd
+         Network
+         .get_staged_ledger_aux_and_pending_coinbases_at_hash_rpcs_received)
+      0. ;
+    Gauge.set (snd Network.answer_sync_ledger_query_rpcs_sent) 0. ;
+    Gauge.set (snd Network.answer_sync_ledger_query_rpcs_received) 0. ;
+    Gauge.set (snd Network.get_transition_chain_rpcs_sent) 0. ;
+    Gauge.set (snd Network.get_transition_chain_rpcs_received) 0. ;
+    Gauge.set (snd Network.get_transition_knowledge_rpcs_sent) 0. ;
+    Gauge.set (snd Network.get_transition_knowledge_rpcs_received) 0. ;
+    Gauge.set (snd Network.get_transition_chain_proof_rpcs_sent) 0. ;
+    Gauge.set (snd Network.get_transition_chain_proof_rpcs_received) 0. ;
+    Gauge.set (snd Network.get_node_status_rpcs_sent) 0. ;
+    Gauge.set (snd Network.get_node_status_rpcs_received) 0. ;
+    Gauge.set (snd Network.get_ancestry_rpcs_sent) 0. ;
+    Gauge.set (snd Network.get_ancestry_rpcs_received) 0. ;
+    Gauge.set (snd Network.ban_notify_rpcs_sent) 0. ;
+    Gauge.set (snd Network.ban_notify_rpcs_received) 0. ;
+    Gauge.set (snd Network.get_best_tip_rpcs_sent) 0. ;
+    Gauge.set (snd Network.get_best_tip_rpcs_received) 0. ;
+    Gauge.set (snd Network.get_epoch_ledger_rpcs_sent) 0. ;
+    Gauge.set (snd Network.get_epoch_ledger_rpcs_received) 0. ;
+    Gauge.set Network.new_state_received 0. ;
+    Gauge.set Network.new_state_broadcasted 0. ;
+    Gauge.set Network.transaction_pool_diff_received 0. ;
+    Gauge.set Network.transaction_pool_diff_broadcasted 0. ;
+    Gauge.set Network.snark_pool_diff_received 0. ;
+    Gauge.set Network.snark_pool_diff_broadcasted 0.) ;
+  Queue.clear Transition_frontier.validated_blocks ;
+  Queue.clear Transition_frontier.rejected_blocks
+
+let start ~logger ~node_status_url ~transition_frontier ~sync_status ~network
+    ~addrs_and_ports ~start_time ~slot_duration =
+  [%log info] "Starting node status service using URL $url"
+    ~metadata:[ ("url", `String node_status_url) ] ;
+  let five_slots = Time.Span.scale slot_duration 5. in
+  reset_gauges () ;
+  every ~start:(after five_slots) ~continue_on_error:true five_slots
+  @@ fun () ->
+  don't_wait_for
+  @@
+  match Broadcast_pipe.Reader.peek transition_frontier with
+  | None ->
+      [%log info] "Transition frontier not available for node status service" ;
+      Deferred.unit
+  | Some tf -> (
+      let catchup_job_states =
+        match Transition_frontier.catchup_tree tf with
+        | Full catchup_tree ->
+            let catchup_states =
+              Transition_frontier.Full_catchup_tree.to_node_status_report
+                catchup_tree
+            in
+            let finished = ref 0
+            and failed = ref 0
+            and to_download = ref 0
+            and to_initial_validate = ref 0
+            and to_verify = ref 0
+            and wait_for_parent = ref 0
+            and to_build_breadcrumb = ref 0 in
+            List.iter catchup_states ~f:(fun (state, n) ->
```

Comment:
Instead of using references and mutation, I think it would be cleaner to use a record type for `finished`, `failed`, etc., then use a `List.fold` where the initial value has 0 for all those fields, and where the fold function uses functional record update on the accumulator, like:
```
match state with
...
| Failed -> { acc with failed = n }
...```